libutils/sequence.c

```diff
@@ -350,3 +350,45 @@ Seq *SeqGetRange(const Seq *seq, size_t start, size_t end)
 
     return sub;
 }
+
+void SeqStringAddSplit(Seq *seq, const char *str, char delimiter)
+{
+    if (str) // TODO: remove this inconsistency, add assert(str)
+    {
+        const char *prev = str;
+        const char *cur = str;
+
+        while (*cur != '\0')
+        {
+            if (*cur == delimiter)
+            {
+                size_t len = cur - prev;
+                if (len > 0)
+                {
+                    SeqAppend(seq, xstrndup(prev, len));
+                }
+                else
+                {
+                    SeqAppend(seq, xstrdup(""));
+                }
+                prev = cur + 1;
+            }
+
+            cur++;
+        }
+
+        if (cur > prev)
+        {
+            SeqAppend(seq, xstrndup(prev, cur - prev));
+        }
```

Comment:
I see now that I misunderstood, I thought the function was used to delimit substrings on one line, but it's only used to delimit lines. Then I agree.